RosettaCore/src/main/java/com/hubspot/rosetta/internal/StoredAsJsonSerializer.java

```diff
@@ -1,18 +1,24 @@
 package com.hubspot.rosetta.internal;
 
+import java.io.IOException;
+import java.lang.reflect.Type;
+
 import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.io.SegmentedStringWriter;
+import com.fasterxml.jackson.core.util.BufferRecycler;
+import com.fasterxml.jackson.databind.BeanProperty;
 import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonSerializer;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.SerializerProvider;
 import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonFormatVisitorWrapper;
+import com.fasterxml.jackson.databind.ser.ContextualSerializer;
 import com.fasterxml.jackson.databind.ser.std.NonTypedScalarSerializerBase;
 import com.fasterxml.jackson.databind.ser.std.StringSerializer;
 
-import java.io.IOException;
-import java.lang.reflect.Type;
-
-public class StoredAsJsonSerializer<T> extends NonTypedScalarSerializerBase<T> {
+public class StoredAsJsonSerializer<T> extends NonTypedScalarSerializerBase<T> implements ContextualSerializer {
```

Comment:
Should we make these same changes to the binary serializer?

Reply:
That's probably a good idea